### PR TITLE
Bug fix to allow SharedGalleryImangeId and CommunityGalleryImageId parameters to New-AzVMConfig

### PR DESCRIPTION
**Pull request by mfortin**
## Description
Adding validation for SharedGalleryImageId and CommunityGalleryImageId in New-AzVM cmdlet

## Mandatory Checklist

- Please choose the target release of Azure PowerShell
  - [X] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] N/A

- [X] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

**Comment by mfortin** [comment]
@microsoft-github-policy-service agree

**Comment by mfortin** [comment]
More context to this PR.

Before this PR:
```powershell
> $VmConfig = New-AzVMConfig -VMName testvm -VMSize Standard_B2s -SharedGalleryImageId /SharedGalleries/<subscription>-<gallery>/Images/<definition/Versions/<version> -Location WestUS2
Write-Output $VmConfig.StorageProfile.ImageReference

Publisher               :
Offer                   :
Sku                     :
Version                 :
ExactVersion            :
SharedGalleryImageId    : /SharedGalleries/<sub>-<gallery>/Images/<definition/Versions/<version>
CommunityGalleryImageId :
Id                      : 

# And then launching a Virtual Machine
> New-AzVM -VM $VmConfig -ResourceGroupName $ResourceGroupName -Location $VmRegion

System.NullReferenceException: Object reference not set to an instance of an object.
   at Microsoft.Azure.Commands.Compute.NewAzureVMCommand.retrieveImageVersion(String publisher, String offer, String sku, String version)
   at Microsoft.Azure.Commands.Compute.NewAzureVMCommand.retrieveSpecificImageFromNotId()
   at Microsoft.Azure.Commands.Compute.NewAzureVMCommand.DefaultExecuteCmdlet()
   at Microsoft.Azure.Commands.Compute.NewAzureVMCommand.ExecuteCmdlet()
   at Microsoft.WindowsAzure.Commands.Utilities.Common.AzurePSCmdlet.ProcessRecord()
System.Management.Automation.InvocationInfo
```

Now, after this PR:
```powershell
$VmConfig = New-AzVMConfig -VMName testvm -VMSize Standard_B2s -SharedGalleryImageId /SharedGalleries/<subscription>-<gallery>/Images/<definition/Versions/<version> -Location WestUS2
Write-Output $VmConfig.StorageProfile.ImageReference

Publisher               :
Offer                   :
Sku                     :
Version                 :
ExactVersion            :
SharedGalleryImageId    : /SharedGalleries/<sub>-<gallery>/Images/<definition/Versions/<version>
CommunityGalleryImageId :
Id                      : 

# And then launching a Virtual Machine
> New-AzVM -VM $VmConfig -ResourceGroupName $ResourceGroupName -Location $VmRegion
# Success!
```

**Comment by mfortin** [comment]
@bilaakpan-ms @Sandido @haagha @grizzlytheodore 
Would you mind reviewing this simple PR ?

**Comment by mfortin** [comment]
Up
@bilaakpan-ms @Sandido @haagha @grizzlytheodore

**Comment by mfortin** [comment]
@bilaakpan-ms @Sandido @haagha @grizzlytheodore
up

**Comment by Sandido** [comment]
Will look into this next week.

**Comment by mfortin** [comment]
@Sandido have you had the chance to have a look by any chance ?

**Comment by Sandido** [comment]
@mfortin , looking at this now, running some tests. In the future, and if you can now, share how you made the SharedGallery.

**Comment by mfortin** [comment]
Hi @Sandido, yes absolutely. This specific gallery was created with the Direct Share feature https://learn.microsoft.com/en-ca/azure/virtual-machines/share-gallery-direct?tabs=portaldirect#prerequisites

**Comment by mfortin** [comment]
```
az sig create \
   --gallery-name myGallery \
   --permissions groups \
   --resource-group myResourceGroup
```

**Comment by Sandido** [comment]
@mfortin do you expect the SharedGalleryImageId to be in this exact format as seen under the resource Properties? /SharedGalleries/<sub>-<gallery>/Images/<definition/Versions/<version>

File: src/Compute/Compute/VirtualMachine/Operation/NewAzureVMCommand.cs [hunk 1]
```diff
@@ -1019,6 +1019,20 @@ public void DefaultExecuteCmdlet()
                         }
                     }
                 }
+                else if (this.VM.StorageProfile?.ImageReference?.SharedGalleryImageId != null) {
```
Comment (Sandido) [inline src/Compute/Compute/VirtualMachine/Operation/NewAzureVMCommand.cs:1022]: for these else if blocks, include the if jobPresent in here with an &&. They're simple enough.

File: src/Compute/Compute/VirtualMachine/Operation/NewAzureVMCommand.cs [hunk 2]
```diff
@@ -1019,6 +1019,20 @@ public void DefaultExecuteCmdlet()
...
+                    // do nothing, send message to use TrustedLaunch.
+                    if (this.AsJobPresent() == false) // to avoid a failure when it is a job. Seems to fail when it is a job.
```
Comment (Sandido) [inline src/Compute/Compute/VirtualMachine/Operation/NewAzureVMCommand.cs:1024]: I've run manual tests on these changes and it works, doesn't break any scenarios.

**Review by Sandido — CHANGES_REQUESTED** [review]

**Comment by Sandido** [comment]
PR is nearly ready for merge, just a couple of tweaks.

File: src/Compute/Compute/ChangeLog.md [hunk 3]
```diff
@@ -35,6 +35,7 @@
 * Added a new parameter `-SharingProfile` to `New-AzCapacityReservationGroup` and `Update-AzCapacityReservationGroup`.
 * Added the new parameter `SourceImageVMId` to the `New-AzGalleryImageVersion` cmdlet. Also added some error messages for this new parameter and the existing parameter `SourceImageId`. 
 * Updated parameter `-TargetRegion` in `New-AzGalleryImageVersion` and `Update-AzGalleryImageVersion` to accept `ExcludeFromLatest` field.
```
Comment (Sandido) [inline src/Compute/Compute/ChangeLog.md:37]: Give this PR a better title.

**Review by Sandido — CHANGES_REQUESTED** [review]

File: src/Compute/Compute/ChangeLog.md (hunk 3 above)
Comment (mfortin) [reply src/Compute/Compute/ChangeLog.md:37]: Updated

File: src/Compute/Compute/VirtualMachine/Operation/NewAzureVMCommand.cs (hunk 1 above)
Comment (mfortin) [reply src/Compute/Compute/VirtualMachine/Operation/NewAzureVMCommand.cs:1022]: I wanted to follow the style of the other if in that condition

Comment (mfortin) [reply src/Compute/Compute/VirtualMachine/Operation/NewAzureVMCommand.cs:1036]: if AsJobPresent prints a warning message, if true, it doesn't print, so it has to stay this way

**Comment by mfortin** [comment]
@sandido Updated as per your review.

File: src/Compute/Compute/ChangeLog.md [hunk 4]
```diff
@@ -22,6 +22,7 @@
 ## Upcoming Release
 * Added `Etag` property to PSVirtualMachine and PSVirtualMachineScaleSet objects.   
 * Added parameters `-IfMatch` and `-IfNoneMatch` to `Update-AzVM`, `Update-AzVmss`, `New-AzVm`, `New-AzVmss`, `New-AzVmConfig`, and `New-AzVmssConfig` cmdlets.
+* Resolves bug with `New-AzVMConfig` for `-CommunityGalleryImageId` and `-SharedGalleryImageId` parameters.
```
Comment (YanaXu) [inline src/Compute/Compute/ChangeLog.md:25]:
```suggestion
* Resolved the bug with `New-AzVMConfig` for `-CommunityGalleryImageId` and `-SharedGalleryImageId` parameters.
```

**Review by YanaXu — APPROVED** [review]

**Pull request merged**